src/components/cards/productCard.js

```diff
@@ -0,0 +1,53 @@
+import './index.css'
+import { useState } from 'react'
+import {useCart,useNavigate} from '../../customHooks'
+import axios from 'axios'
+const ProductCard=({productObj})=>{
+  const{price,imageUrl,title,description,productRating,readyToDeliver}=productObj
+
+  const {dispatchCart,wishlistCount,cartCount}=useCart()
+  
+  const navigate=useNavigate()
+    const [inCart,setButtonText]=useState(false)
+
+    const addItemToCart=async(product)=>{
+      try{
+        console.log(product,"entered intotry")
+        const response=await axios.post('/api/user/cart',{product},{
+          headers:{
+            authorization:"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI3MTUxN2NmOS05MTg4LTRlNGYtOWM1MS0xMzMxZWE1ZThkZmQiLCJlbWFpbCI6ImFkYXJzaGJhbGlrYUBnbWFpbC5jb20ifQ.WTkYnS_dAUXq8sBn-GKoX0BC6ZJKNpL8Q_CNUzlebJI"
+          }
+        });
+        setButtonText(true)
+        dispatchCart({type:"cart",payload:{cart:response.data.cart,cartCount:cartCount+1}})
```

Comment:
give this type a good name like SET_TO_CART or something like that and again do not use it in a hardcoded manner instead get it from an object and like ACTIONS and you can access all the type values in dispatch throughout your app.

Reply:
ok wil do this